### DISCUSSION TITLE
Fetching All Records Regardless of Pagination in Grid View

### DISCUSSION THREAD

**Discussion by EdwinThomas11** [serenity-is/Serenity#7158] [Help] [answered]
Hello Team,

`this.view.getItems()` retrieves only the grid values based on the current pagination. However, I need to get all the records, regardless of the defined pagination. Is there a function available to fetch all the records, or do I need to make an API call to get the values?

Thanks in advance.

**Comment by VictorTomaili** [comment, marked as answer]
The view stores the current page records only. If you need all records, you need to make an api call. If you prefer to remove pagination completely, and show all records in the grid, you can override the `usePager` and return false like `protected usePager() { return false; }` in your grid. Then you can use getItems to retrieve all records.

**Reply by EdwinThomas11** [reply]
Thanks for the reply. I currently have `usePager` set to false, which results in slow loading when there are many records. To avoid this slowness, I need to implement pagination. Therefore, I will use an API call instead.

**Reply by EdwinThomas11** [reply]
@VictorTomaili  Is it possible to override `this.view.getItems()` so that I can centralize my API call, instead of having to update it in multiple locations?

**Reply by VictorTomaili** [reply]
I would not recommend doing this.

Setting `usePager = false` causes the grids to attempt loading all data from the database without filters, leading to slow load times. Similarly, modifying the getItems function can result in the same issue. The getItems function is part of sleekgrid, and altering it without understanding its internal workings within the sleekgrid library may lead to unintended consequences. 

Additionally, making API calls can also lead to slow load times. It may be necessary to reevaluate your approach and consider caching the API response for repeated use.

**Reply by EdwinThomas11** [reply]
Thank you for your response; it is very helpful.
